Add Backdrops, Behang and National Geographic sources

TheLastZombie opened the pull request:
- [Backdrops](https://backdrops.io/) via [API](https://backdrops.io/walls/api_v3.2.php?task=all_walls)
- [Behang](http://knokfirst.com/behang/) via [API](https://knokfirst.com/behang_manifest.json)
- [National Geographic](https://www.nationalgeographic.co.uk/photo-of-day) via [API](https://www.nationalgeographic.co.uk/page-data/photo-of-day/page-data.json) (fixes #339)

sidml commented:
@peterlevi 
Will be really nice if this can be merged.

TheLastZombie commented:
FYI: You can already use these by placing the Python files in `~/.config/variety/plugins/downloaders`.

peterlevi commented:
Hi all. @TheLastZombie, thanks a lot for the contribution!

I'm evaluating these sources for adding as default sources, in relation to the section "Image sources in Variety
" at https://peterlevi.com/variety/2017/03/changes/.

Sorry, but Backdrops is a no-go: there is no attribution or legal info on their website, but from what I gather that's a collection of images they have hand-crafted to feed into their own Android app. The API is not really a public API, but just some endpoint they've opened to their own app. I don't see it proper to include in Variety.

It's a similar thing with Behang - it's a collection of images someone created, no proper author attribution, no legal/copyright info on the website.

For these two sources, I'd be fine to merge only if we first manage to somehow get in touch with the site owners and have an official permission from them to include in Variety. If you're willing to drive this communication, please CC me in the emails.

The National Geographic source seems like one we can add, cause it does provide author credits in the metadata:
![Screenshot from 2022-04-22 10-47-46](https://user-images.githubusercontent.com/1457048/164642948-be9eda61-e1a9-430b-88fd-b782de166a8e.png)

But we'll need to fetch and populate those credits into the downloaded image, along with the caption. Author name, link to author page, and caption are the things I see we should save into the image. Please see UnsplashDownloader and BingDownloader for examples how to populate this sort of metadata into the downloaded image.

peterlevi commented on variety/plugins/builtin/downloaders/NationalGeographicDownloader.py:
```diff
@@ -0,0 +1,62 @@
+        url = item["entity"]["mediaImage"]["url"]
+        origin_url = NationalGeographicDownloader.ROOT_URL + "?image=" + splitext(basename(url))[0]
+        image_url = "https://static.nationalgeographic.co.uk" + url
+        return self.save_locally(origin_url, image_url)
```
Let's use the `extra_metadata` parameter to set `author`, `authorURL` and `description` fro mthe provided data in the json. Please see https://github.com/varietywalls/variety/blob/ba157a5aad81a55bff024dffcee7e719cc1fd1b4/variety/plugins/builtin/downloaders/UnsplashDownloader.py#L116 as an example.

peterlevi reviewed (changes requested):
Please see my comments. Let's separate out the NG source, address comments for it for fetching author credits, and we can merge it. The other two sources are a no-go without some communication with the site owners.

TheLastZombie commented:
Thanks for the thorough tutorial! I removed the Backdrops and Behang sources. I also added the extra metadata to the National Geographic source - somewhat. Two problems arose:

- The author URL/`path.alias` paths don't actually seem to go anywhere. Taking `/photographer/jim-richardson`, for example, none of the following URLs exist:
  - https://www.nationalgeographic.co.uk/photographer/jim-richardson
  - https://www.nationalgeographic.co.uk/photo-of-day/photographer/jim-richardson
  - https://www.nationalgeographic.com/photographer/jim-richardson
  Not even [searching Google](https://www.google.com/search?q=inurl%3A%2Fphotographer%2Fjim-richardson) results in anything that makes sense, and there are no actual links to an author page on the [Photo of the Day page](https://www.nationalgeographic.co.uk/photo-of-day) either.

- I still added author and description to the best of my judgement, but I couldn't actually find where these fields are displayed in Variety, so I couldn't actually test them.

peterlevi commented:
Thanks, @TheLastZombie . I just added a Google search link for "authorURL" - having both author and authorURL are necessary so that we show the "Author" info in the menu. Now merged to master. 

Again, thanks a lot for the contribution!
